River segments (based on river valley)

fnattino opened the pull request:
And here is the result of the corridor segmentation based on the river valley. 

Note that there is some potential for performance improvement, since we probably can consider only a fraction of the network edges for the continuity analysis, namely the ones that intersects the river corridor.

There is also need for some post-processing, for the block that does not intersects the waterway.

![Screenshot 2024-10-07 at 11 41 27](https://github.com/user-attachments/assets/48c4b067-d247-4c7d-939b-32080ebb3779)

cforgaci reviewed (commented):
Looks good! As discussed yesterday, please try the following:

1. Merge polygons that are not crossed by the river with the smallest neighbouring polygon.
2. Merge polygons that do not cross the corridor with the smallest neighbouring corridor-crossing polygon. In a generalised version, this might need to be done in a recursive way.

fnattino commented:
Hi @cforgaci , I have updated the notebook in two ways:
* I have slightly reduced the threshold in the angle to grow strokes using the modified COINS approach, which helps in creating more candidate boundaries between segments (I have lowered the threshold from 120 to 100).
* I have implemented the strategy to merge segments as you described above. 

Here is an overview of the final result - I think it looks good!

![Screenshot 2024-10-09 at 16 58 47](https://github.com/user-attachments/assets/4ed71eed-216e-461f-8ba1-d6cf45ad6d16)

cforgaci commented:
> Here is an overview of the final result - I think it looks good!

Yes, it does! Thanks, @fnattino. Feel free to merge.

One thing we can change in a separate PR is the capping. Following the morphological approach to delineation, it makes more sense to follow the same approach when capping too. I suggest we cap by default with a shortest path between the valley edge end points rather then using the admin boundary.
